clang/test/CodeGenSYCL/kernel-handler.cpp

```diff
@@ -0,0 +1,38 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple nvptx64-unknown-unknown -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s --check-prefix=NONATIVESUPPORT
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s --check-prefix=NATIVESUPPORT
+
+// This test checks IR generated when kernel_handler argument
+// (used to handle SYCL 2020 specialization constants) is passed
+// by kernel
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+
+void test(int val) {
+  queue q;
+  q.submit([&](handler &h) {
+    int a;
+    kernel_handler kh;
+    h.single_task<class test_kernel_handler>(
+        [=](auto) {
+          int local = a;
+        },
+        kh);
+  });
+}
+
+// NONATIVESUPPORT: define dso_local void @"_ZTSZZ4testiENK3$_0clERN2cl4sycl7handlerEE19test_kernel_handler"
```

Comment:
Does the test pass on Windows?  May need to wildcard the mangled names.

Reply:
We don't need to since we specify the target triple right?

Reply:
Does nvptx or spir imply Linux somehow?  I really don't know what happens with the third 'unknown'.   If it passes on Windows, you may be right.

Reply:
I'm not sure. Since I don't have a Windows system set up to test this, I added wildcard just to be safe. Thanks for review!